kani-driver/src/call_cbmc.rs

```diff
@@ -164,6 +184,94 @@ impl KaniSession {
     }
 }
 
+impl VerificationResult {
+    /// Computes a `VerificationResult` (kani-driver's notion of the result of a CBMC call) from a
+    /// `VerificationOutput` (cbmc_output_parser's idea of CBMC results).
+    ///
+    /// NOTE: We actually ignore the CBMC exit status, in favor of two checks:
+    ///   1. Examining the actual results of CBMC properties.
+    ///       (CBMC will regularly report "failure" but that's just our cover checks.)
+    ///   2. Positively checking for the presence of results.
+    ///       (Do not mistake lack of results for success: report it as failure.)
+    fn from(output: VerificationOutput, start_time: Instant) -> VerificationResult {
+        if let Some(mut items) = output.processed_items {
+            let result_idx = items.iter().position(|x| matches!(x, ParserItem::Result { .. }));
+            if let Some(result_idx) = result_idx {
+                let result = items.remove(result_idx);
+                if let ParserItem::Result { result } = result {
```

Comment:
This is quite hard to read. Can we refactor into small functions (find, extract) so it's easier to read?